plane/src/drone/runtime/unix_socket/mod.rs

```diff
@@ -0,0 +1,145 @@
+            .client
+            .send_request(MessageToServer::Prepare(config.clone()))
+            .await?;
+        match response {
+            MessageToClient::PrepareResult(Ok(())) => Ok(()),
+            MessageToClient::PrepareResult(Err(e)) => Err(Error::msg(e)),
+            _ => Err(Error::msg("Unexpected response from server")),
+        }
+    }
+
+    async fn spawn(
+        &self,
+        backend_id: &BackendName,
+        executable: DockerExecutorConfig,
+        acquired_key: Option<&AcquiredKey>,
+        static_token: Option<&BearerToken>,
+    ) -> Result<SpawnResult> {
+        let response = self
+            .client
+            .send_request(MessageToServer::Spawn(
+                backend_id.clone(),
+                executable,
+                acquired_key.cloned(),
+                static_token.cloned(),
+            ))
+            .await?;
+        match response {
+            MessageToClient::SpawnResult(Ok(result)) => Ok(result),
+            MessageToClient::SpawnResult(Err(e)) => Err(Error::msg(e)),
+            _ => unreachable!("Unexpected response from server"),
+        }
+    }
+
+    async fn terminate(&self, backend_id: &BackendName, hard: bool) -> Result<bool> {
+        let response = self
+            .client
+            .send_request(MessageToServer::Terminate(backend_id.clone(), hard))
+            .await?;
+        match response {
+            MessageToClient::TerminateResult(Ok(result)) => Ok(result),
+            MessageToClient::TerminateResult(Err(e)) => Err(Error::msg(e)),
+            _ => unreachable!("Unexpected response from server"),
+        }
+    }
+
+    fn events(&self) -> impl Stream<Item = TerminateEvent> + Send {
+        let mut event_rx = self.client.subscribe_events();
+        Box::pin(async_stream::stream! {
+            while let Ok(event) = event_rx.recv().await {
+                if let MessageToClient::TerminateEvent(terminate_event) = event {
+                    yield terminate_event;
+                }
+            }
+        })
+    }
+
+    fn metrics_callback<F: Fn(BackendMetricsMessage) + Send + Sync + 'static>(&self, sender: F) {
+        let mut event_rx = self.client.subscribe_events();
+        tokio::spawn(async move {
+            while let Ok(event) = event_rx.recv().await {
+                if let MessageToClient::MetricsMessage(metrics) = event {
+                    sender(metrics);
+                }
+            }
+        });
+    }
+
+    async fn wait_for_backend(
+        &self,
+        backend: &BackendName,
+        address: SocketAddr,
+    ) -> Result<(), BackendError> {
+        let response = self
+            .client
+            .send_request(MessageToServer::WaitForBackend(backend.clone(), address))
+            .await
+            .expect("Failed to send request");
+        match response {
+            MessageToClient::WaitForBackendResult(Ok(())) => Ok(()),
+            MessageToClient::WaitForBackendResult(Err(e)) => Err(e),
```

Comment:
If this is intentional to be explicit I'm fine with it, but why not:

```rust
MessageToClient::WaitForBackendResult(v) => v,
```

to cover both cases?